### ISSUE TITLE
Support VMSS and mixed VMSS and VM

### ISSUE THREAD

**Issue by Flowman** [kubernetes-sigs/azuredisk-csi-driver#868]
**Is your feature request related to a problem?/Why is this needed**

Support VMSS, right now it only looks for virtual machines. This is bit of an issue in a k3s setup where the masters are VMs but all the worker nodes are VMSS.

**Describe the solution you'd like in detail**

Support mixed environment where VM and VMSS are used to build a cluster.

**Additional context**

I0528 05:56:31.926187       1 azure_vmclient.go:132] Received error in vm.get.request: resourceID: /subscriptions/xxx/resourceGroups/rg/providers/Microsoft.Compute/virtualMachines/xxx, error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 404, RawError: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 404, RawError: {"error":{"code":"ResourceNotFound","message":"The Resource 'Microsoft.Compute/virtualMachines/xxx' under resource group 'rancher-rg' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix"}}
I0528 05:56:31.926394       1 azure_wrap.go:194] Virtual machine "xxx" not found

**Comment by andyzhangx** [comment]
can you set `"vmType": "vmss"` in cloud config?

https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/372d5b0fda6faf900564f423113e3d5be0f741dc/deploy/example/azure.json#L12

**Comment by Flowman** [comment]
Guess I cannot have a generic kubernetes _azure-cloud-provider_ secret for all host.

Will try with a azure.json on each host and see how it goes.

**Comment by andyzhangx** [comment]
> Guess I cannot have a generic kubernetes _azure-cloud-provider_ secret for all host.
> 
> Will try with a azure.json on each host and see how it goes.

looks like a new requirement, agent node and master nodes needs different `azure-cloud-provider` config.

**Comment by Flowman** [comment]
Just an example of an k3s cluster

NAME                       STATUS   ROLES                       AGE     VERSION
k3s-vmss000000      Ready    worker                      3h33m   v1.20.7+k3s1
k3s-vmss000001      Ready    worker                      3h33m   v1.20.7+k3s1
k3s-master               Ready    control-plane,etcd,master   3h35m   v1.20.7+k3s1

**Comment by andyzhangx** [comment]
if your controller is running on `k3s-master`, you should set `"vmType": "vmss"` in cloud config, not need to care about master node since you only need to attach/detach disk to agent node.

**Comment by Flowman** [comment]
I use the helm chart and it installs on all nodes.

**Comment by andyzhangx** [comment]
try add `--set controller.runOnMaster=true`, it would run csi-azuredisk-controller on master node

**Comment by Flowman** [comment]
the controller is fine, it should be more --set node.runOnMaster=false for the daemonset in csi-azuredisk-node.yaml

**Comment by andyzhangx** [comment]
> si-azuredisk-node.yaml

it's ok to run csi-azuredisk-node on agent node, so this issue is resolved?

**Comment by Flowman** [comment]
I noticed some issue running both VMSS and VM. If the controller runs on a VMSS and tires to create a file share on a VM it will error out with wrong VM type.

So I would not recommend running mixed mode of VMSS and VM.

I opened a issue on the azurefile csi, but will suggest it here to. This will help with azuredisk node not installing on the master.

**Is your feature request related to a problem?/Why is this needed**


It would be good if we could change tolerations for the csi-azuredisk-node.yaml. As is would be nice to control if it should be created on a master node or not, where it will never be used.

- operator: "Exists" installs on all server in the cluster.

**Describe the solution you'd like in detail**


Add a chart value to change tolerations for nodes.
